bdk_chain/src/keychain/keychain_txout_index.rs

```diff
@@ -237,21 +238,24 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
             .script_pubkeys()
             .get(&index)
             .expect("we just added it");
-        (next_derivation_index, new_spk)
+        (next_derivation_index, new_spk.to_owned())
     }
 
-    /// Gets the next usued script pubkey in the keychain i.e. the script pubkey with the lowest index that has not been used yet.
+    /// Gets the next unused script pubkey in the keychain i.e. the script pubkey with the lowest index that has not been used yet.
     ///
     /// ## Panics
     ///
     /// Panics if `keychain` has never been added to the index
-    pub fn derive_next_unused(&mut self, keychain: &K) -> (u32, &Script) {
-        let need_new = self.keychain_unused(keychain).next().is_none();
-        // this rather strange branch is needed because of some lifetime issues
-        if need_new {
-            self.derive_new(keychain)
+    pub fn next_unused<'a>(&'a mut self, keychain: &K) -> (u32, Script) {
+        let existing = {
+            self.keychain_unused(keychain)
+                .next()
+                .map(|(i, s)| (i, s.to_owned()))
+        };
```

Comment:
NIT: We don't need these extra brackets.

Reply:
I didn't get it.. Which extra brackets?